Add ATOM, IRIS

yurii-khi opened the pull request:
Known issues, todo:
 - [x] handle `rpc_urls` field for `coins_config.json` generation


Refs:
https://github.com/KomodoPlatform/air_dex/pull/677
https://github.com/KomodoPlatform/air_dex/pull/571#issuecomment-1256944049

smk762 commented:
wrt: https://github.com/KomodoPlatform/coins/issues/518 we should add `"gui_auth": true` to the node entries for quicknode servers.

tonymorony commented on utils/coins_config.json:
```diff
@@ -10833,6 +10836,8 @@
         ]
     },
+=======
+>>>>>>> master
```
? @smk762

smk762 replied:
coin config gen failed due to missing fields in `coins` file. I've added those, but might need swap_contract (if applicable) and still have to integrate them into the script so they are included in output.

cipig commented on api_ids/coingecko_ids.json:
```diff
@@ -31,15 +31,17 @@
     "ATOM-BEP20": "cosmos",
+    "ATOM-IBC_IRIS": "cosmos-hub",
     "ATOM-PLG20": "cosmos",
+    "ATOM": "cosmos-hub",
```
API id for coingecko is "cosmos", see https://www.coingecko.com/en/coins/cosmos-hub
![image](https://user-images.githubusercontent.com/32116761/198897197-cbd08c4b-333b-4bba-b10d-9b20029022cc.png)

cipig commented on api_ids/coingecko_ids.json:
```diff
@@ -259,6 +261,7 @@
     "IOTA-BEP20": "iota",
     "IOTX-BEP20": "iotex",
     "IOTX-PLG20": "iotex",
+    "IRIS": "irisnet",
```
gecko_id is "iris-network"
![image](https://user-images.githubusercontent.com/32116761/198897267-2aa5b76b-50d5-4cb9-ae4c-ee8fe7054b9b.png)
https://www.coingecko.com/en/coins/irisnet

smk762 commented:
I've fixed up the config generator to include Tendermint & tokens. @yurii-khi please advise if there is anything missing or incorrect in https://github.com/KomodoPlatform/coins/blob/add-atom-iris/utils/coins_config.json#L20036-L20098

Not sure there was meant to be a `swap_contract_address` for these? I added files for COSMOS / tCOSMOS & IRIS / tIRIS to https://github.com/KomodoPlatform/coins/tree/add-atom-iris/ethereum also, though I had no info for tCOSMOS rpc_node url so it's just empty for future use.

yurii-khi commented:
Considering https://github.com/KomodoPlatform/atomicDEX-API/pull/1522, reverted changes to BNB, BNBT, MATIC, and ETH `swap_contract_address`.

yurii-khi commented:
> I've fixed up the config generator to include Tendermint & tokens. @yurii-khi please advise if there is anything missing or incorrect in https://github.com/KomodoPlatform/coins/blob/add-atom-iris/utils/coins_config.json#L20036-L20098
> 
> Not sure there was meant to be a `swap_contract_address` for these? I added files for COSMOS / tCOSMOS & IRIS / tIRIS to https://github.com/KomodoPlatform/coins/tree/add-atom-iris/ethereum also, though I had no info for tCOSMOS rpc_node url so it's just empty for future use.

It looks pretty good for me, thank you so much for your help, @smk762 !
Tried generated `coins_config.json` here https://github.com/KomodoPlatform/air_dex/pull/677, and it works just fine. Only thing I changed manually is naming for 'nodes' field, so I'd like to ask if can rename 'nodes' to 'rpc_urls' for ATOM, IRIS, and ATOM-IBC_IRIS?
And also I'm not sure if we should place them into `ethereum` directory, are they really ethereum networks?
PS: regarding  `swap_contract_address` for IRIS - we don't need it afaik.

smk762 commented:
Yeah I thought no swap contract address. They're not really "ethereum networks" tho neither is QTUM within the same folder. Perhaps we should rename the folder to something like `networks`, though not sure if that might break things for anyone. Since the other folder is `electrum` we could also maybe call this one `enable` so it matches the methods they are associated with.
Simple enough to change nodes -> rpc_urls, and since cosmos not in desktop yet it wont ripple out to disrupt anything

tonymorony commented on ethereum/IRIS:
```diff
@@ -1,8 +1,8 @@
     {
-      "url": "https://iris.komodo.live/"
+      "url": "https://iris.komodo.live/",
+      "gui_auth": true
```
@smk762 why there is `"gui_auth"` flag for IRIS/ATOM nodes? (afaik its only needed for our quicknode infra setup)

smk762 replied:
will remove, I thought they were quicknode urls

cipig commented on coins:
```diff
@@ -11450,5 +11450,54 @@
+    }
+  },
+  {
+    "coin": "ATOM-IBC_IRIS",
```
afaik the only special char allowed in ticker is `-`... and it is used to separate the actual ticker from the chain name... price service uses ticker to fetch prices from Binance
what is the chain name of this? we already have BEP20, ERC20, PLG20 and so on

yurii-khi replied:
Afaik the chain name is actually `IBC_IRIS`, at least we handle it like [this](https://github.com/KomodoPlatform/air_dex/blob/9b670d6b16e8f0c3241335fb7fb78bf256f372bb/lib/shared/utils/utils.dart#L210) on gui side. Also api currently accepts `ATOM-IBC_IRIS` as coin (tested with https://github.com/KomodoPlatform/atomicDEX-API/commit/004d5caabd5b3383068e8c7d4e114188bd16a2ca), so I assume `_` character is now allowed as well.

cipig replied:
ok, but why so complicated, why not simply `ATOM-IBC` or `ATOM-IRIS` or so?

smk762 replied:
I asked a similar Q in a meeting last week - there could be additions in future which are IBC but not IRIS (e.g. ATOM-KMD_IBC)

tonymorony commented:
@smk762 

```
Conflicting files
api_ids/coingecko_ids.json
api_ids/coinpaprika_ids.json
utils/electrum_scan_report.json
```
